### PR TITLE
add reset-trace

### PR DESCRIPTION
**Pull request by brendangregg**
this came up in #763

### CODE REVIEW

**Comment by palmtenor** [comment]
Thanks! @drarmstr and I encountered a bunch of such issues when developing our tools. We wrote something our own but it would be good to have something available to public.

File: tools/reset-trace.sh [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+done
+shift $(( $OPTIND - 1 ))
+
+### reset tracing state
+vecho "Reseting tracing state..."
+vecho
+cd $tracing || die "ERROR: accessing tracing. Root user? /sys/kernel/debug?"
+writefile kprobe_events
+writefile uprobe_events
+writefile trace         # clears trace_pipe
```
Comment (4ast) [inline tools/reset-trace.sh]: what about adding something like:
echo nop > current_tracer
echo nobin > trace_options
echo context-info > trace_options

**Comment by brendangregg** [comment]
@4ast Yes, I had them in my original ftrace one, but stripped them out, on the notion that if we wern't setting them, then why reset them (in case someone is using them). This is what I also had:

```
writefile current_tracer nop
writefile set_ftrace_filter
writefile set_graph_function
writefile set_ftrace_pid
writefile events/enable 0
writefile tracing_thresh 0
writefile tracing_on 1
```

Where there is no argument, it's a "> file".

**Comment by 4ast** [comment]
makes sense.
can we warn that the options are not in the state that bcc expects?
All bpf_trace_printk examples will dump junk when trace_pipe is in binary mode...

**Comment by brendangregg** [comment]
ok, so you mean not adding them back makes sense? :)

I'll add warnings if it finds it needs to reset something, and change cat to "cat -v" to handle binary.

**Comment by 4ast** [comment]
yes. I mean it makes sense not to 'reset' the global state that bcc scripts didn't touch :)
may be separate script for expected bcc global state is needed... or a flag to this reset-trace... though warn about unexpected global configuration is probably sufficient.

**Comment by brendangregg** [comment]
Updated. It'll now warn on detecting unrelated (ftrace) files are in use, and can reset them with -F for force.

**Comment by 4ast** [comment]
nice!

**Pull request merged**